core-bundle/src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -917,11 +917,20 @@ public static function resolveFlaggedUrl(&$url)
 	 */
 	public static function convertEncoding($str, $to, $from=null)
 	{
-		if (!$str)
+		if (!static::isStringable($str))
 		{
+			@trigger_error('Passing a non-stringable argument to StringUtil::convertEncoding() has been deprecated an will no longer work in Contao 5.0.', E_USER_DEPRECATED);
+
 			return '';
 		}
 
+		$str = static::toString($str);
```

Comment:
```suggestion
		$str = (string) $str;
```

We already know that `$str` is stringable at this point, therefore we do not need a private method for the cast.

Reply:
The idea was to provide the utility methods `toString` and `isStringable` for other use cases. However, as they are private now and only used in `convertEncoding`, they might be just inlined.

The  `isStringable` check should be inlined as well then (and may get extracted to a utility method in the future if any other method requires a similar check).

I will adjust the code accordingly.